tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,359 @@
+    noise_skip_neighborfreqs - int
+        set this >=1 if you want to exclude the immediately neighboring
+        frequency bins in noise level calculation
+
+    Returns
+    -------
+    snr - np.array
+        array containing snr for all epochs, channels, frequency bins.
+        NaN for frequencies on the edge, that do not have enoug neighbors on
+        one side to calculate snr
+    """
+
+    # prep not epoched / single channel data
+    is_2d = True if (len(psds.shape) == 2) else False
+    if is_2d:
+        psds = psds.reshape((1, psds.shape[0], psds.shape[1]))
+
+    # SNR loop
+    snr = np.empty(psds.shape)
+    for i_freq in range(psds.shape[2]):
+
+        # skip freqs on the edges (without noise neighbors)
+
+        start_freq_i = noise_n_neighborfreqs + noise_skip_neighborfreqs
+        stop_freq_i = (psds.shape[2] - noise_n_neighborfreqs
+                       - noise_skip_neighborfreqs)
+        if not (stop_freq_i > i_freq >= start_freq_i):
+            snr[:, :, i_freq] = np.nan
+            continue
+
+        # extract signal level
+        signal = psds[:, :, i_freq]
+
+        # ... and average noise level
+        i_noise = []
+        for i in range(noise_n_neighborfreqs):
+            i_noise.append(i_freq + noise_skip_neighborfreqs + i + 1)
+            i_noise.append(i_freq - noise_skip_neighborfreqs - i - 1)
+        noise = psds[:, :, i_noise].mean(axis=2)
+
+        snr[:, :, i_freq] = signal / noise
+
+    # reshape not epoched / single channel data to original dimensions
+    if is_2d:
+        snr = snr.reshape(snr.shape[1], snr.shape[2])
+
+    return snr
+
+
+###############################################################################
+# Calculate SNR
+stim_freq = 12.
+noise_n_neighborfreqs = 3
+noise_skip_neighborfreqs = 1
+snrs = snr_spectrum(psds, noise_n_neighborfreqs=noise_n_neighborfreqs,
+                    noise_skip_neighborfreqs=noise_skip_neighborfreqs)
+
+###############################################################################
+# Find frequency bin containing stimulation frequency - ideally, with the exact
+# stimulation frequency in the center.
+tmp_distlist = abs(np.subtract(freqs, stim_freq))
+i_signal = np.where(tmp_distlist == min(tmp_distlist))[0][0]
+# could be updated to support multiple frequencies
+
+###############################################################################
+# Extract and average SNRs at this frequency
+snrs_stim = snrs[:, :, i_signal]  # trial subselection can be done here
+print('average SNR at %iHz (all channels, all trials): %.3f '
+      % (stim_freq, snrs_stim.mean()))
+
+
+###############################################################################
+# Visualization
+# -------------
+
+
+##############################################################################
+# Plot power spectral density
+# code snippet from
+# https://martinos.org/mne/stable/auto_examples/time_frequency/plot_compute_raw_data_spectrum.html  # noqa E501
```

Comment:
heres the `# noqa..` again, and it appears in the built html

Reply:
That's an interesting one. We need `# noqa` here to keep flake8 happy but somewhere between sphinx and sphinx-gallery, they don't treat it as a comment, well, because it is not when it is a part of another comment. Not sure what to do about it. Let's keep it for now.

Reply:
alright, lets keep this in mind..